### PR TITLE
add mod to Tensor

### PR DESCRIPTION
**Pull request by geohotstan**
tests passed by splitting the kernel with contiguous 🤔 
I guess some rewrite is rewriting this part `r + b * (((r < 0) & (b > 0)) | ((r > 0) & (b < 0)))` away
gonna stare at this a bit more and maybe practice my VIZ skillz...

  
> possibily can be cleaner with a floor div

something like this without the new `class Mod(Function):`?
```python
def mod(self, x:Union[Tensor, ConstType], reverse=False) -> Tensor:
  a, b = self._broadcasted(x, reverse)
  return a - b * (a/b).floor().cast(self.dtype)
```

or by fixing `def __floordiv__(self, x): return self.idiv(x)  # TODO: idiv is trunc div, not floordiv`?

### CODE REVIEW

**Comment by geohotstan** [comment]
disabling this rule makes it pass without contiguous

`(UPat(GroupOp.ALU, name="x"), lambda x: x.const_like(x.vmin) if x.vmin == x.vmax else None),` [here](https://github.com/tinygrad/tinygrad/blob/master/tinygrad/ops.py#L1232)

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -3129,6 +3129,20 @@ def div(self, x:Union[Tensor, ConstType], reverse=False) -> Tensor:
...
+    Mod `self` by `x`.
+    Equivalent to `self % x`.
+    Supports broadcasting to a common shape, type promotion, and integer inputs.
+
+    ```python exec="true" source="above" session="tensor" result="python"
+    print(Tensor([-4, 7, 5, 4, -7, 8]).mod(Tensor([2, -3, 8, -2, 3, 5])).numpy())
+    ```
+    """
+    a, b = self._broadcasted(x, reverse)
+    r = F.Mod.apply(a, b).contiguous()
```
Comment (geohot) [inline tinygrad/tensor.py:3143]: Why is this contiguous needed?

Comment (geohotstan) [reply tinygrad/tensor.py:3143]: https://github.com/chenyuxyz/tinygrad/pull/636
without it tests fail.

`(UPat(GroupOp.ALU, name="x"), lambda x: x.const_like(x.vmin) if x.vmin == x.vmax else None)`
This is the rule that's causing problems.

**Comment by chenyuxyz** [comment]
see #8458, should unblock you and thanks for root causing the issue!

**Pull request closed without merging**